.github/workflows/tests.yml

```diff
@@ -15,6 +15,63 @@ on:
       - '!docs/**'
       - '!*.md'
 jobs:
+  tests-using-nix:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-24.04]
+        type: [qemu-user-tests, qemu-tests, tests]
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    steps:
+    - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # @v3
+    - uses: cachix/install-nix-action@08dcb3a5e62fa31e2da3d490afc4176ef55ecd72  # @v30
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+
+    - name: Install dependencies
+      run: |
+        ./setup-dev.sh --install-only
```

Comment:
You could enter the `nix develop` environment instead to also ensure all the tests are working from nix-installed dependencies. 

Assuming `nix develop` there is https://github.com/DeterminateSystems/magic-nix-cache which I believe would help with caching things like custom jemalloc derivation, etc.

Reply:
Unfortunately if we run tests from `nix develop` they don't work well because `gcc` (from nixpkgs) overwrites libc.so to `/nix/store/..libc.so`, and we have a lot of tests that would need to be fixed because of that.

Reply:
In next step we could try to fix these tests